src/Control/Controller.php

```diff
@@ -737,4 +751,12 @@ public static function get_template_global_variables()
             'CurrentPage' => 'curr',
         ];
     }
+
+    protected function getTemplateEngine(): TemplateEngine
+    {
+        if (!$this->templateEngine) {
+            $this->templateEngine = Injector::inst()->create(TemplateEngine::class);
```

Comment:
```suggestion
            $this->templateEngine = Injector::inst()->get(TemplateEngine::class);
```
Should be getting a singleton here?

Reply:
Why?

Reply:
There's only needs to be a single instance of a template engine correct? If so just seems like we should be using a singleton for that. If the engine has setters on it (now or in the future) then presumbably we'd want the engines state to be updated "globally"?

Reply:
The constructor can take template candidates. In the case of our engine there's also a bunch of instance properties that get set during rendering which are only reset during instantiation. Different engines may also have other per-instance requirements.
It's easier to develop and maintain the engines if they don't have to function as singletons.